### PR TITLE
Remove unwrap in client find_session

### PR DESCRIPTION
**Pull request by ab-cgh**

### CODE REVIEW

File: rustls/src/client/hs.rs [hunk 1]
```diff
@@ -77,21 +77,22 @@ fn find_session(
...
-    let key_buf = key.get_encoding();
-
-    let maybe_value = sess
-        .config
-        .session_persistence
-        .get(&key_buf);
+    let key_buf = {
+        let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
+        key.get_encoding()
+    };
```
Comment (briansmith) [inline rustls/src/client/hs.rs:83]: Seems like this change to the scope of `key` is out of scope since it isn't needed for removing the `unwrap()` and it isn't clearly better (or worse) than what was there before. I would revert this change and maybe add this to a TODO for discussing code style changes.

File: rustls/src/client/hs.rs [hunk 2]
```diff
@@ -77,21 +77,22 @@ fn find_session(
...
 
-    if maybe_value.is_none() {
-        debug!("No cached session for {:?}", dns_name);
-        return None;
-    }
+    let config = &sess.config.session_persistence;
+  
+    let value = match config.get(&key_buf) {
```
Comment (briansmith) [inline rustls/src/client/hs.rs:87]: I would have made this change instead:
```
    let value = sess
        .config
        .session_persistence
        .get(&key_buf)
        .or_else(|| {
            debug!("No cached session for {:?}", dns_name);
            None
        })?;
```

File: rustls/src/client/hs.rs [hunk 3]
```diff
@@ -77,21 +77,22 @@ fn find_session(
...
+        Some(v) => v,
+        None => {
+            debug!("No cached session for {:?}", dns_name);
+            return None;
+        }
+    };
 
-    let value = maybe_value.unwrap();
-    let mut reader = Reader::init(&value[..]);
```
Comment (briansmith) [inline rustls/src/client/hs.rs:94]: switching from `&value[..]` to `value.as_slice()` is also an out-of-scope change that should be reverted.

If we want to switch the coding pattern here, we should probably do so globally. Probably we should just do what Clippy recommends. Note that `as_slice()` is also unneeded here; just `&value` is enough. But, let's save that for another day.

**Review by briansmith — CHANGES_REQUESTED** [review]

File: rustls/src/client/hs.rs (hunk 1 above)
Comment (ab-cgh) [reply rustls/src/client/hs.rs]: This has been reverted.

File: rustls/src/client/hs.rs (hunk 3 above)
Comment (ab-cgh) [reply rustls/src/client/hs.rs:94]: Reverted for now as well.

File: rustls/src/client/hs.rs (hunk 2 above)
Comment (ab-cgh) [reply rustls/src/client/hs.rs]: Sure, this works as well. I made the change to keep less intermediate variables.

**Review by briansmith — APPROVED** [review]

**Review by djc — APPROVED** [review]
Looks good, thanks!
